maven-core/src/main/java/org/apache/maven/plugin/internal/DefaultPluginDependenciesResolver.java

```diff
@@ -109,34 +110,8 @@ public Artifact resolve(Plugin plugin, List<RemoteRepository> repositories, Repo
             request.setTrace(trace);
             ArtifactDescriptorResult result = repoSystem.readArtifactDescriptor(pluginSession, request);
 
-            if (result.getDependencies() != null) {
-                for (org.eclipse.aether.graph.Dependency dependency : result.getDependencies()) {
-                    if ("org.apache.maven".equals(dependency.getArtifact().getGroupId())
-                            && "maven-compat".equals(dependency.getArtifact().getArtifactId())
-                            && !JavaScopes.TEST.equals(dependency.getScope())) {
-                        pluginValidationManager.reportPluginValidationIssue(
-                                session,
-                                pluginArtifact,
-                                "Plugin depends on the deprecated Maven 2.x compatibility layer, which may not be supported in Maven 4.x");
-                    }
-                }
-
-                Set<String> mavenArtifacts = result.getDependencies().stream()
-                        .filter(d -> !JavaScopes.PROVIDED.equals(d.getScope()) && !JavaScopes.TEST.equals(d.getScope()))
-                        .map(org.eclipse.aether.graph.Dependency::getArtifact)
-                        .filter(a -> "org.apache.maven".equals(a.getGroupId()))
-                        .filter(a -> !MavenPluginDependenciesValidator.EXPECTED_PROVIDED_SCOPE_EXCLUSIONS_GA.contains(
-                                a.getGroupId() + ":" + a.getArtifactId()))
-                        .filter(a -> a.getVersion().startsWith("3."))
-                        .map(a -> a.getGroupId() + ":" + a.getArtifactId() + ":" + a.getVersion())
-                        .collect(Collectors.toSet());
-
-                if (!mavenArtifacts.isEmpty()) {
-                    pluginValidationManager.reportPluginValidationIssue(
-                            session,
-                            pluginArtifact,
-                            "Plugin should declare these Maven artifacts in `provided` scope: " + mavenArtifacts);
-                }
+            for (MavenPluginDependenciesValidator dependenciesValidator : dependenciesValidators) {
+                dependenciesValidator.validate(session, pluginArtifact, result);
```

Comment:
I think that dependency list returned by `repoSystem.readArtifactDescriptor` is different than `repoSystem.collectDependencies`

But for building plugin classloder we use output of `resolveInternal` method ... so maybe we should check dependencies returned by `repoSystem.collectDependencies`

Reply:
readArtifactDescriptor == POM
resolveInternal = collection and resolving of POM (that above) augmented with things like WagonExcluder, ExclusionsDependencyFilter (excludes core and extension exported artifacts).

So, first is kinda "raw" while second is "processed", and due exclusions, 2nd will have core stuff excluded always,

Reply:
Not every artifacts is excluded, pleas look: https://gist.github.com/slawekjaranowski/c65612fca4094ec8a2d09c6db3b8f78a

like this:

```
[DEBUG] Populating class realm plugin>org.apache.maven.plugins:maven-war-plugin:3.4.0-SNAPSHOT
...
[DEBUG]   Included: org.sonatype.sisu:sisu-inject-bean:jar:1.4.2
[DEBUG]   Included: org.sonatype.sisu:sisu-guice:jar:noaop:2.1.7
[DEBUG]   Included: org.codehaus.plexus:plexus-component-annotations:jar:1.5.5
[DEBUG]   Included: org.sonatype.plexus:plexus-sec-dispatcher:jar:1.3
[DEBUG]   Included: org.sonatype.plexus:plexus-cipher:jar:1.4
...
```

Reply:
I know that every core artifacts are stripped ... 
but we check if plugin has such on dependency list - so we have two different source of checked and used dependencies 

In my example we have:
```
[DEBUG]    org.apache.maven.shared:maven-mapping:jar:3.0.0:compile
[DEBUG]       org.apache.maven:maven-core:jar:3.0:compile
```
but validator do not detect that `maven-core` is in compile scope on dependencies

Reply:
@slawekjaranowski unsure what dependency plugin uses to display tree, but (to me) it seems it lacks information. Here is the 'verbose" tree that current resolver emits (in verbose mode): Don't be mislead by war version, as there is java snippet as well, I manipulate war POM BEFORE collect (basically making maven-core dep provided in war 3.3.2.

My point is, that "enhanced" DependencyGraphDumper (since 1.9.8) prints out extra info (more than dep plugin). 

This is the "verbose" (and dirty) tree as there was no graph transformation applied). The fact that line has `(xxx)` at ends, means that node **is eliminated**, but is left for verbosity.

So we may want to look into here, but IMHO there is no "two sources" and the dep-tree seems to lack information...

Reply:
@cstamas - ok, let's leave as is - we can improve later if needed